edge-hub/core/src/Microsoft.Azure.Devices.Edge.Hub.Core/storage/MessageStore.cs

```diff
@@ -261,28 +261,38 @@ private async Task CleanQueue(bool checkEntireQueueOnCleanup)
                 {
                     foreach (KeyValuePair<string, ISequentialStore<MessageRef>> endpointSequentialStore in this.messageStore.endpointSequentialStores)
                     {
+                        var messageQueueId = endpointSequentialStore.Key;
                         try
                         {
                             if (this.cancellationTokenSource.IsCancellationRequested)
                             {
                                 return;
                             }
 
-                            Events.CleanupTaskStarted(endpointSequentialStore.Key);
-                            CheckpointData checkpointData = await this.messageStore.checkpointStore.GetCheckpointDataAsync(endpointSequentialStore.Key, CancellationToken.None);
+                            var (endpointId, priority) = MessageQueueIdHelper.ParseMessageQueueId(messageQueueId);
+                            Events.CleanupTaskStarted(messageQueueId);
+                            CheckpointData checkpointData = await this.messageStore.checkpointStore.GetCheckpointDataAsync(messageQueueId, CancellationToken.None);
                             ISequentialStore<MessageRef> sequentialStore = endpointSequentialStore.Value;
-                            Events.CleanupCheckpointState(endpointSequentialStore.Key, checkpointData);
+                            Events.CleanupCheckpointState(messageQueueId, checkpointData);
                             int cleanupEntityStoreCount = 0;
 
+                            // If cleanup just peek head, message counts is tailOffset-headOffset+1
+                            // otherwise count while iterating over the queue.
+                            var headOffset = 0L;
+                            var tailOffset = sequentialStore.GetTailOffset(CancellationToken.None);
+                            var messageCounts = 0L;
```

Comment:
nit: messageCount ?

Reply:
done

Reply:
we can probably just remove the class member to reduce naming confusion

---
In reply to: [525486168](https://github.com/Azure/iotedge/pull/3980#discussion_r525486168) [](ancestors = 525486168)